crates/protocol/src/batch/type.rs

```diff
@@ -0,0 +1,77 @@
+//! Batch Types
+//!
+//! This module contains the batch types for the OP Stack derivation pipeline.
+//!
+//! ## Batch
+//!
+//! A batch is either a `SpanBatch` or a `SingleBatch`.
+//!
+//! The batch type is encoded as a single byte:
+//! - `0x00` for a `SingleBatch`
+//! - `0x01` for a `SpanBatch`
+
+use alloy_rlp::{Decodable, Encodable};
+
+/// The single batch type identifier.
+pub const SINGLE_BATCH_TYPE: u8 = 0x00;
+
+/// The span batch type identifier.
+pub const SPAN_BATCH_TYPE: u8 = 0x01;
+
+/// The Batch Type.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[repr(u8)]
+pub enum BatchType {
+    /// Single Batch.
+    Single = SINGLE_BATCH_TYPE,
+    /// Span Batch.
+    Span = SPAN_BATCH_TYPE,
+}
+
+impl From<u8> for BatchType {
+    fn from(val: u8) -> Self {
+        match val {
+            SINGLE_BATCH_TYPE => Self::Single,
+            SPAN_BATCH_TYPE => Self::Span,
+            _ => panic!("Invalid batch type: {val}"),
+        }
+    }
+}
+
+impl From<&[u8]> for BatchType {
+    fn from(buf: &[u8]) -> Self {
+        Self::from(buf[0])
+    }
+}
```

Comment:
is this used anywhere? imo no point given there's `From<u8>`, so this should be handled where its used instead